comp/otelcol/ddprofilingextension/impl/extension.go

```diff
@@ -87,13 +107,39 @@ func (e *ddExtension) startForOCB() error {
 		profiler.WithAPIKey(string(e.cfg.API.Key)),
 	)
 
-	// todo(mackjmr): add datadogexporter hostmetadata provider to retrieve hostname, and
-	// pass it to profiler via profiler.WithHostname(). This requires a refactor in contrib,
-	// as the logic lives within an internal package.
 	if string(e.cfg.API.Site) != "" {
 		profilerOptions = append(profilerOptions, profiler.WithSite(string(e.cfg.API.Site)))
 	}
 
+	source, err := e.sourceProvider.Source(context.Background())
+	if err != nil {
+		return err
+	}
+
+	var tags strings.Builder
+	tags.WriteString(fmt.Sprintf("agent_version:%s", version.AgentVersion))
+	if e.cfg.ProfilerOptions.Env != "" {
+		tags.WriteString(fmt.Sprintf(",default_env:%s", e.cfg.ProfilerOptions.Env))
+	}
+
+	if source.Kind == "host" {
+		profilerOptions = append(profilerOptions, profiler.WithHostname(source.Identifier))
+		tags.WriteString(fmt.Sprintf(",host:%s", source.Identifier))
+	}
+
+	if source.Kind == "task_arn" {
+		tags.WriteString(fmt.Sprintf(",orchestrator:fargate_ECS,task_arn:%s", source.Identifier))
```

Comment:
I would try to follow our implementation in profiles.go, which lowercases the orchestrator name (which could also be `fargate_eks`:
https://github.com/DataDog/datadog-agent/blob/067029a5c7b6ca88688f52be8ec68618a6313874/pkg/trace/api/profiles.go#L83

Reply:
I've lowercased it. Unfortunately, our hostname resolution does not support EKS fargate yet, only ECS fargate. I'm working on adding support for this (https://datadoghq.atlassian.net/browse/OTAGENT-316), and will make a PR to update this code once it's supported.

Reply:
Cool, thanks for the update!